File: back/src/main/kotlin/com/jvmvstv_v/back/user/service/UserService.kt
package com.jvmvstv_v.back.user.service

import com.jvmvstv_v.back.user.model.LoginInput
import com.jvmvstv_v.back.user.model.RegisterInput
import com.jvmvstv_v.back.user.model.UpdateProfileInput
import com.jvmvstv_v.back.user.model.User
import com.jvmvstv_v.back.user.model.UserRole
import com.jvmvstv_v.back.user.model.UserStatus
import java.util.UUID

interface UserService {
    fun currentUser(): User?
    fun findById(id: UUID): User?
    fun updateProfile(input: UpdateProfileInput): User
    fun register(input: RegisterInput): User
    fun login(input: LoginInput): User
    fun logout()
    fun listUsers(): List<User>
    fun setUserStatus(userId: UUID, status: UserStatus): User
    fun setUserRole(userId: UUID, role: UserRole): User
}

File: front/src/services/user/UserService.ts
import type { LoginInput, RegisterInput, UpdateProfileInput, User, UserRole, UserStatus } from './types.ts'

export interface UserService {
  me(): Promise<User | null>
  getUser(id: string): Promise<User | null>
  updateProfile(input: UpdateProfileInput): Promise<User>
  updateAvatar(mimeType: string, imageBase64: string): Promise<User>
  register(input: RegisterInput): Promise<User>
  login(input: LoginInput): Promise<User>
  logout(): Promise<void>
  confirmEmail(code: string): Promise<User>
  resendVerificationCode(): Promise<void>
  users(): Promise<User[]>
  setUserStatus(userId: string, status: UserStatus): Promise<User>
  setUserRole(userId: string, role: UserRole): Promise<User>
}
